src/scores/probability/brier_impl.py

```diff
@@ -61,3 +67,155 @@ def brier_score(
+        ValueError: if ``obs`` contains the dimension 'threshold'.
+        ValueError: if ``weights`` contains the dimension 'threshold'.
+        ValueError: if ``ensemble_member_dim`` is 'threshold'.
+        ValueError: if ``ensemble_member_dim`` is not in ``fcst`` dimensions.
+
+    References:
+        - Ferro, C. A. T. (2013). Fair scores for ensemble forecasts. Quarterly
+            Journal of the Royal Meteorological Society, 140(683), 1917–1923.
+            https://doi.org/10.1002/qj.2270
+
+    See Also:
+        - :py:func:`scores.probability.brier_score`
+
+    Examples:
+        Calculate the Brier score for an ensemble forecast for a single threshold:
+
+        >>> import numpy as np
+        >>> import xarray as xr
+        >>> from scores.probability import ensemble_brier_score
+        >>> fcst = xr.DataArray(np.random.rand(10, 10), dims=['time', 'ensemble'])
+        >>> obs = xr.DataArray(np.random.rand(10), dims=['time'])
+        >>> ensemble_brier_score(fcst, obs, ensemble_member_dim='ensemble', thresholds=0.5)
+
+        Calculate the Brier score for an ensemble forecast for multiple thresholds:
+        >>> thresholds = [0.1, 0.5, 0.9]
+        >>> ensemble_brier_score(fcst, obs, ensemble_member_dim='ensemble', thresholds=thresholds)
+
+    """
+    if threshold_mode not in [">=", ">"]:
+        raise ValueError("threshold_mode must be either '>=' or '>'.")
+    if ensemble_member_dim == "threshold":
+        raise ValueError("The ensemble_member_dim is not allowed to be 'threshold'.")
+    if "threshold" in fcst.dims:
+        raise ValueError("The dimension 'threshold' is not allowed in fcst.")
+    if "threshold" in obs.dims:
+        raise ValueError("The dimension 'threshold' is not allowed in obs.")
+    if ensemble_member_dim not in fcst.dims:
+        raise ValueError("ensemble_member_dim must be one of the dimensions in fcst.")
+
+    weights_dims = None
+    if weights is not None:
+        if "threshold" in weights.dims:
+            raise ValueError("The dimension 'threshold' is not allowed in weights.")
+        weights_dims = weights.dims
+
+    dims_for_mean = gather_dimensions(
+        fcst.dims,
+        obs.dims,
+        weights_dims=weights_dims,
+        reduce_dims=reduce_dims,
+        preserve_dims=preserve_dims,
+        score_specific_fcst_dims=ensemble_member_dim,
+    )
+    if isinstance(thresholds, (float, int)):
+        thresholds = [thresholds]
+    thresholds_xr = xr.DataArray(thresholds, dims=["threshold"], coords={"threshold": thresholds})
+
+    # calculate i term in equation
+    member_event_count = (fcst >= thresholds_xr).sum(dim=ensemble_member_dim)
+    # calculate m term in equation
+    total_member_count = fcst.notnull().sum(dim=ensemble_member_dim)
+
+    binary_obs = binary_discretise(obs, thresholds, threshold_mode)
+
+    result = (member_event_count / total_member_count - binary_obs) ** 2
+    if fair_correction:
```

Comment:
The docs said something about this only happening when the number of ensemble members was not 1 (presumably when > 1)  but this condition enters the code block in all cass when fair_correction is set to true, so you are relying on the semantics of the maths below.

Reply:
I might have to improve the docstring if it was confusing.

The desired behaviour is:
- Only apply the fair correction if `fair_correction=True`
- The exception is when a NaN is produced by a divide by zero error. This is handled with  `fair_correction = fair_corr.fillna(0)`

Reply:
The quote from the docstring that I'm referencing is 

> If ``fair_correction`` is set to ``True`` and the number of ensemble members is 1, the fair correction will not be > applied to avoid division by zero.

The way I read this is:
 a) If fair correction is 'true" and also the number of ensemble members is != 1, the fair correction will be applied
 b) If the fair correction is "false" and the number of ensemble members == 1, the fair correction will be applied
 c) If both of the fair correction is 'True' together with the number of ensemble members ==1, then the fair correction will not be applied (and never otherwise, see a. and b.)
 d) The purpose of not applying the fair correction is to avoid a div by zero which occurs when the number of ensemble members is zero

The actual logic seems to be that if fair correction is "true", then regardless of the number of ensemble members (i.e. whether it equals 1 or not), the fair correction is applied.

Another approach would be to raise an exception when the fair is set to true and the number of ensemble members is 1. Consider whether it is more appropriate to put the onus back onto the user rather than making an assumption about what they want, particularly if they were perhaps unaware or mistaken in suppling only on ensemble member. Consider also raising a user warning when this occurs to alert the user to the situation.

Reply:
e.g. the logic in the code is not `if fair_correction and len(ensemblle_members) ==1: ...`

Reply:
Okay, I think I understand the issue. Perhaps you interpreted the "if" as an "if and only if". I'll have a go at rewording it to avoid confusion.

The statement b) should be false.

The fair correction is only applied when `fair_correction==True` and at coordinates where the number of ensemble members is > 1.

> The actual logic seems to be that if fair correction is "true", then regardless of the number of ensemble members (i.e. whether it equals 1 or not), the fair correction is applied.

No, it's not applied when the ensemble member count = 1. This is tested with the input data `DA_FCST_ENS` where `stn=105`

It may not be obvious where I implemented this logic, but it occurs on the `fair_correction = fair_corr.fillna(0)` line
In this situation, if `fair_correction` = True, then for any data point where there is a single ensemble member, `fair_corr` will have NaNs for those coords, and so `fair_correction` will have 0 values for those coords due to the `fair_correction = fair_corr.fillna(0)` .  I think that I'll have to add a comment to make this clearer in the code as it's not obvious.

I've set this up so that it will work if you have a lagged ensemble and there is at least one timestep where there is only a single ensemble member.

Reply:
Thanks, your latest docstring and code comment are much more readable (for me anyway!)